### DISCUSSION TITLE
For windows, Eloquent DB:: with sqlite3.exe?

### DISCUSSION THREAD

**Discussion by osc2nuke** [NativePHP/laravel#190] [Ideas]
When using the `DB::` query's that Laravel provide, i noticed how slow the app really was.
In this case, import a CSV with 115.000+ rows +/- 28MB, what took between 10 and sometimes even 20 minutes.
I was able to hook-on to the sqlite.exe via Laravel on my machine, where it only took 2 seconds!

Here is how i did it (tvview is the Laravel project location)

In one of my Controller function:
```
        $db = "C:/wamp64/www/tvview/database/nativephp.sqlite";//location of the database
        $sql = "C:/wamp64/www/tvview/storage/app/import.sql";//sql that wil be executed
        exec("C:/wamp64/www/tvview/database/sqlite3.exe ".$db."<".$sql);
//I Copied sqlite.exe from my C:\ and putted it in my project folder.
//call sqlite3.exe
// connect to $db:
//execute the $sql

```
The content inside import.sql that is going to be executed via exec():
```
CREATE TABLE IF NOT EXISTS "channels" ("id" varchar not null, "group_title" varchar, "tvg_name" varchar, "tvg_logo" varchar, "group_type" varchar, "url" varchar, "slug" varchar, "favorite" varchar, primary key ("id"));
.mode csv
.import C:/wamp64/www/tvview/storage/app/output.csv channels
```
Is there some package already existing i could use what would make this easier, especially now that i also perform a search via `DB::` , what takes ages?
Or could this be a future feature/option in NativePHP.
I do not have any knowledge how this would work, or is even possible for Linux and Mac os.

Thanks in advance!

PS: I spend over a week to find out about above, wasted a lot of time with the given answer(s)
[ HERE: Laravel and SQLite dot commands](https://laracasts.com/discuss/channels/laravel/laravel-and-sqlite-dot-commands)

As sqlite3 simply does not support .dot commands, only via the command-line interface

**Comment by simonhamp** [comment]
You'd have to be able to ship platform-specific builds of the executable (in this case SQLite) with your app. It's certainly not out of the realm of possibility, but it will require very specific interaction with the runtime (Electron/Tauri) to make sure it packages these extra binaries appropriately.

This doesn't feel like a problem that NativePHP should be solving at this time. Maybe in a future version we'll have a way to make this easier.
